src/main/java/at/klausbetz/provider/AppleIdentityProviderEndpoint.java

```diff
@@ -58,19 +57,24 @@ public AppleIdentityProviderEndpoint(AppleIdentityProvider appleIdentityProvider
 
     @POST
     public Response authResponse(@FormParam(OAUTH2_PARAMETER_STATE) String state, @FormParam(OAUTH2_PARAMETER_CODE) String authorizationCode, @FormParam(OAUTH2_PARAMETER_USER) String user, @FormParam(OAuth2Constants.ERROR) String error) {
+        if (state == null) {
+            return errorIdentityProviderLogin(Messages.IDENTITY_PROVIDER_MISSING_STATE_ERROR);
+        }
+
         IdentityBrokerState idpState = IdentityBrokerState.encoded(state, realm);
         String clientId = idpState.getClientId();
         String tabId = idpState.getTabId();
-
         if (clientId == null || tabId == null) {
             logger.errorf("Invalid state parameter: %s", state);
-            event.event(EventType.LOGIN);
-            event.error(Errors.IDENTITY_PROVIDER_LOGIN_FAILURE);
-            return ErrorPage.error(session, null, Response.Status.BAD_REQUEST, Messages.INVALID_REQUEST);
+            return errorIdentityProviderLogin(Messages.INVALID_REQUEST, Response.Status.BAD_REQUEST);
         }
+
+        AuthenticationSessionModel authSession = this.callback.getAndVerifyAuthenticationSession(state);
+        session.getContext().setAuthenticationSession(authSession);
+
         if (error != null) {
             logger.error(error + " for broker login " + appleIdentityProvider.getConfig().getProviderId());
```

Comment:
Could you please lower the log level here to debug?  
Sure, I did this back then. However, it's no fatal error and a warn should still make clear something went wrong.

Reply:
yes sure